File: app/Http/Controllers/SpecialistPatternsJava/Creational/Prototype/PrototypeJava/src/ColorPoint.java
public class ColorPoint extends Point implements Prototype {
    String color;

    // Конструктор класса ColorPoint, принимающий координаты x, y и цвет.
    public ColorPoint(int x, int y, String color) {
        super(x, y);
        this.color = color;
    }

    // Конструктор класса ColorPoint, создающий копию существующей точки ColorPoint.
    public ColorPoint(ColorPoint p) {
        this(p.x, p.y, p.color);
    }

    // Метод Clone() для создания клона (копии) текущего объекта ColorPoint.
    @Override
    public ColorPoint Clone() {
        return new ColorPoint(this);
    }

    // Переопределенный метод toString() для вывода информации о ColorPoint в виде строки.
    @Override
    public String toString() {
        return String.format("Color Point (%d,%d) %s", x, y, color);
    }
}

File: app/Http/Controllers/SpecialistPatterns/Prototype/ColorPoint.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\Prototype;

// Класс "Цветная точка", наследуется от класса "Точка"
class ColorPoint extends Point {
    private $color;

    // Конструктор класса, инициализирует координаты x и y, а также цвет.
    public function __construct($x, $y, $color) {
        parent::__construct($x, $y);
        $this->color = $color;
    }

    // Метод clone() создает и возвращает клон (копию) объекта цветной точки.
    public function clone() {
        return new ColorPoint($this->x, $this->y, $this->color);
    }

    // Переопределение метода __toString() для получения строкового представления объекта.
    public function __toString() {
        return "Color Point ({$this->x}, {$this->y}) {$this->color}";
    }
}
